zio-schema-protobuf/shared/src/main/scala/zio/schema/codec/FieldMappingCache.scala

```diff
@@ -0,0 +1,37 @@
+package zio.schema.codec
+
+import scala.collection.mutable
+
+import zio.schema.Schema
+
+/**
+ * A per-encooding/decoding cache for field mappings. No need for thread safety as a single encoding/decoding
+ * is sequential.
+ */
+class FieldMappingCache {
```

Comment:
Maybe worth making the constructor private so you have to go through `FieldMapping`?

Reply:
Too much Rust coding, I though it's private because there is no `pub` :D